### PR TITLE
Record the user who last changed the Scan Profile

### PR DESCRIPTION
**Pull request by originalsouth**
### Changes

Record the user who last changed the Scan Profile

### Issue link

None

### Demo
```
{
  "type": "ScanProfile",
  "level": 2,
  "reference": "URL|internet|https://mispo.es/",
  "scan_profile_type": "declared",
  "user_id": "superuser@localhost",
  "xt/id": "ScanProfile|URL|internet|https://mispo.es/"
}
```

### QA notes

Create a bunch of objects and raise their clearance level and run:
```
octopoes/tools/xtdb-cli.py query '{:query {:find [(pull ?var [*])] :where [[?var :type "ScanProfile"][?var :user_id ?user_id][(not= ?user_id nil)]]}}' | jq -C | less -R
```

---

### Code Checklist

- [x] All the commits in this PR are properly PGP-signed and verified.
- [x] This PR only contains functionality relevant to the issue.
- [ ] I have written unit tests for the changes or fixes I made.
- [x] I have checked the documentation and made changes where necessary.
- [x] I have performed a self-review of my code and refactored it to the best of my abilities.

- [x] Tickets have been created for newly discovered issues.
- [x] For any non-trivial functionality, I have added integration and/or end-to-end tests.
- [x] I have informed others of any required `.env` changes files if required and changed the `.env-dist` accordingly.
- [x] I have included comments in the code to elaborate on what is not self-evident from the code itself, including references to issues and discussions online, or implicit behavior of an interface.

---

## Checklist for code reviewers:

Copy-paste the checklist from [the docs/source/templates folder](https://github.com/minvws/nl-kat-coordination/blob/main/docs/source/templates/pull_request_template_review_code.md) into your comment.

---

## Checklist for QA:

Copy-paste the checklist from [the docs/source/templates folder](https://github.com/minvws/nl-kat-coordination/blob/main/docs/source/templates/pull_request_template_review_qa.md) into your comment.

### CODE REVIEW

File: octopoes/octopoes/models/__init__.py [hunk 1]
```diff
@@ -77,6 +77,7 @@ class ScanProfileBase(BaseModel):
     scan_profile_type: str
     reference: Reference
     level: ScanLevel
+    user: str | None = None
```
Comment (underdarknl) [inline octopoes/octopoes/models/__init__.py:80]:
```suggestion
    user: int | None = None
```

File: rocky/account/mixins.py [hunk 2]
```diff
@@ -136,7 +136,7 @@ def verify_raise_clearance_level(self, level: int) -> bool:
     def raise_clearance_level(self, ooi_reference: Reference, level: int) -> bool:
         self.verify_raise_clearance_level(level)
         self.octopoes_api_connector.save_scan_profile(
-            DeclaredScanProfile(reference=ooi_reference, level=ScanLevel(level)),
+            DeclaredScanProfile(reference=ooi_reference, level=ScanLevel(level), user=str(self.request.user)),
```
Comment (underdarknl) [inline rocky/account/mixins.py:139]:
```suggestion
            DeclaredScanProfile(reference=ooi_reference, level=ScanLevel(level), user=self.request.user.id),
```

File: rocky/account/mixins.py [hunk 3]
```diff
@@ -145,7 +145,10 @@ def raise_clearance_level(self, ooi_reference: Reference, level: int) -> bool:
     def raise_clearance_levels(self, ooi_references: list[Reference], level: int) -> bool:
         self.verify_raise_clearance_level(level)
         self.octopoes_api_connector.save_many_scan_profiles(
-            [DeclaredScanProfile(reference=reference, level=ScanLevel(level)) for reference in ooi_references],
+            [
+                DeclaredScanProfile(reference=reference, level=ScanLevel(level), user=str(self.request.user))
```
Comment (underdarknl) [inline rocky/account/mixins.py:149]:
```suggestion
                DeclaredScanProfile(reference=reference, level=ScanLevel(level), user=self.request.user.id)
```

**Comment by originalsouth** [comment]
image

File: rocky/rocky/views/scan_profile.py [hunk 4]
```diff
@@ -19,6 +20,20 @@ class ScanProfileDetailView(FormView, OOIDetailView):
...
+                    context["username"] = str(kat_user)
+                else:
+                    name = str(kat_user)
+                    context["username"] = "".join(
+                        [c + "\u0336" if i < len(name) - 1 else c for i, c in enumerate(name)]
+                    )
+            except get_user_model().DoesNotExist:
+                context["username"] = ""
+        else:
+            context["username"] = ""
```
Comment (underdarknl) [inline rocky/rocky/views/scan_profile.py:36]:
```suggestion
        context["scan_profile_user"] = False
        if self.ooi.scan_profile and self.ooi.scan_profile.user:
            try:
                context["scan_profile_user"] = get_user_model().objects.get(id=self.ooi.scan_profile.user)
            except get_user_model().DoesNotExist:
                pass
```

File: rocky/rocky/templates/scan_profiles/scan_profile_detail.html [hunk 5]
```diff
@@ -12,7 +12,11 @@
                 {% include "partials/ooi_head.html" with ooi=ooi view="scan_profile_detail" %}
 
                 <h2 rf-selector="clearance-header">{% translate "Current clearance level" %}</h2>
-                <p>{{ ooi.scan_profile.human_readable }}, {{ ooi.scan_profile.scan_profile_type }}</p>
+                {% if username %}
+                    <p>{{ ooi.scan_profile.human_readable }}, {{ ooi.scan_profile.scan_profile_type }} (by user "{{ username }}")</p>
```
Comment (underdarknl) [inline rocky/rocky/templates/scan_profiles/scan_profile_detail.html:16]:
```suggestion
                    <p>{{ ooi.scan_profile.human_readable }}, {{ ooi.scan_profile.scan_profile_type }} (by user "{% if not scan_profile_user.is_active %}<s>{{ scan_profile_user }}</s>{% else %} {{ scan_profile_user }}{% endif %}")</p>
```

File: rocky/rocky/views/scan_profile.py [hunk 6]
```diff
@@ -19,6 +20,12 @@ class ScanProfileDetailView(FormView, OOIDetailView):
     def get_context_data(self, **kwargs) -> dict[str, Any]:
         context = super().get_context_data(**kwargs)
         context["mandatory_fields"] = get_mandatory_fields(self.request)
+        context["scan_profile_user"] = False
```
Comment (dekkers) [inline rocky/rocky/views/scan_profile.py:23]: `{% if scan_profile_user %}` also works if the variable does not exist and will just evaluate to False, so it is not necessary to add this.

```suggestion
```

File: rocky/rocky/templates/scan_profiles/scan_profile_detail.html [hunk 7]
```diff
@@ -12,7 +12,19 @@
                 {% include "partials/ooi_head.html" with ooi=ooi view="scan_profile_detail" %}
 
                 <h2 rf-selector="clearance-header">{% translate "Current clearance level" %}</h2>
-                <p>{{ ooi.scan_profile.human_readable }}, {{ ooi.scan_profile.scan_profile_type }}</p>
+                {% if username %}
```
Comment (dekkers) [inline rocky/rocky/templates/scan_profiles/scan_profile_detail.html:15]:
```suggestion
                {% if scan_profile_user %}
```

**Comment by TwistMeister** [comment]
## Checklist for QA:

- [x] I have checked out this branch, and successfully ran a fresh `make reset`.
- [x] I confirmed that there are no unintended functional regressions in this branch:
  - [x] I have managed to pass the onboarding flow
  - [x] Objects and Findings are created properly
  - [x] Tasks are created and completed properly
- [x] I confirmed that the PR's advertised `feature` or `hotfix` works as intended.
- [x] I checked the logs for errors and/or warnings and made issues where necessary

### What works:
image

**Review by ammar92 — APPROVED** [review]

**Review by underdarknl — APPROVED** [review]

**Pull request merged**